src/v1/middlewares/validators/validator.ts

```diff
@@ -0,0 +1,34 @@
+// validateResource.ts
+import { Request, Response, NextFunction } from 'express';
+import { z, AnyZodObject } from 'zod';
+import { responseStatusCodes } from '../../helpers/common';
+import {
+  VALIDATION_ERROR,
+  INTERNAL_SERVER_ERROR,
+} from '../../helpers/locale.json';
+
+const validateResource =
+  (schema: AnyZodObject) =>
+  (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Please add the return type since we are using typescript.
If this is going to prepare a response of the API then use IReturnType.

See src\v1\types\common.ts

```
export interface IReturnType {
  success: boolean;
  data: { resType?: number; resData: any };
  error?: unknown;
  [key: string]: unknown;
}
```

Reply:
Middleware nature: In Express middleware, the focus is on either calling next() or sending a response. The return value of middleware isn't typically used in the rest of the pipeline, so the return type being void has no side effects.